services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortPropertiesSpec.groovy

```diff
@@ -0,0 +1,251 @@
+package org.openkilda.functionaltests.spec.switches
+
+import static org.openkilda.functionaltests.extension.tags.Tag.VIRTUAL
+import static org.openkilda.model.PortProperties.DISCOVERY_ENABLED_DEFAULT
+import static org.openkilda.testing.Constants.NON_EXISTENT_SWITCH_ID
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.SwitchChangeType
+import org.openkilda.northbound.dto.v2.switches.PortPropertiesDto
+import org.openkilda.testing.service.northbound.NorthboundServiceV2
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.See
+
+@See("https://github.com/telstra/open-kilda/tree/develop/docs/design/network-discovery/disable-port-discovery")
+@Narrative("""Some switch ports should not be used in network discovery process.
+By default all ports on all switches are available for discovery.
+Admin has ability to enable/disable discovery on a specific port on a switch using Northbound REST API.
+
+This spec assumes that port discovery property is enabled for all available ports.
+""")
+class PortPropertiesSpec extends HealthCheckSpecification {
+    @Autowired
+    NorthboundServiceV2 northboundV2
+
+    def "Able to manipulate with port properties"() {
+        given: "A port with port properties"
+        // can't use `getAllowedPortsForSwitch` for virtual env in this test,
+        // portProperties validate port number(port number should be in list of '/api/v1/switches/:switch-id/ports')
+        def isl = topology.notConnectedIsls.first()
+        with(northboundV2.getPortProperties(isl.srcSwitch.dpId, isl.srcPort)) {
+            it.switchId == isl.srcSwitch.dpId
+            it.portNumber == isl.srcPort
+            it.discoveryEnabled == DISCOVERY_ENABLED_DEFAULT
+        }
+
+        when: "Update port discovery property"
+        PortPropertiesDto newPortProperties = new PortPropertiesDto()
+        def newDiscoveryEnabled = false
```

Comment:
Well now this should be the opposite to `DISCOVERY_ENABLED_DEFAULT`. Otherwise if the default is changed to 'false' the test will do nothing.